MergerLogic/DataTypes/Gpkg.cs

```diff
@@ -73,26 +74,48 @@ public override void Reset()
             this._offset = 0;
```

Comment:
Please add lock on this line in case `Reset` will be called in a thread in the future.
In addition, maybe changing `this._offset` get and set functions to use a lock will be best? see: https://stackoverflow.com/questions/7161413/thread-safe-properties-in-c-sharp